ai-service/app/tests/rag/test_llm.py

```
@@ -0,0 +1,35 @@
+        mock_groq.chat.completions.create = AsyncMock(return_value=mock_response)
+
+        client = LLMClient()
+        result = asyncio.run(client.generate([{"role": "user", "content": "Hi"}]))
+
+        assert result == "Hello world"
+
+    def test_generate_raises_on_error(self, mock_groq):
+        mock_groq.chat.completions.create = AsyncMock(side_effect=Exception("API error"))
+
+        client = LLMClient()
+        with pytest.raises(Exception):
```

Ruff (0.16.2)

[warning] Do not assert blind exception: `Exception`

(B017)